destination/writer/writer.go

```diff
@@ -282,10 +281,16 @@ func (s *SnowflakeCSV) CheckTable(ctx context.Context, operation sdk.Operation,
 	// table not found
 	if !res.Next() {
 		sdk.Logger(ctx).Info().Msgf("table %s does not exist yet", s.TableName)
+
 		return nil
 	}
 
-	//nolint:gosec // not an issue
+	if err = res.Err(); err != nil {
+		sdk.Logger(ctx).Err(err).Msg("error occurred while checking table rows")
+
+		return errors.Errorf("error occurred while checking table rows: %w", err)
```

Comment:
My general advice would be the following (we even have it in our [guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md#error-handling)):

> Any error needs to be handled by either logging the error and recovering from it, or wrapping and returning it to the caller. We should never both log and return the error.

This way we prevent logging the same error multiple times.

Reply:
@lovromazgon addressed, thank you https://github.com/conduitio-labs/conduit-connector-snowflake/pull/119/commits/b964c37c002eec5457c215f30814632669eef97d